File: core/src/main/java/com/larryhsiao/nyx/core/attachments/Attachment.java
package com.larryhsiao.nyx.core.attachments;

/**
 * Attachment of a Jot
 */
public interface Attachment {
    /**
     * Id of this attachment.
     */
    long id();

    /**
     * The Jot id that this attachment file attached to.
     */
    long jotId();

    /**
     * Uri of this attachment
     */
    String uri();

    /**
     * Version of this attachment item.
     */
    int version();

    /**
     * Indicates if this attachment is deleted or not.
     */
    boolean deleted();
}

File: core/src/main/java/com/larryhsiao/nyx/core/attachments/Attachment.kt
package com.larryhsiao.nyx.core.attachments

/**
 * Attachment of a Jot
 */
interface Attachment {
    /**
     * Id of this attachment.
     */
    fun id(): Long

    /**
     * The Jot id that this attachment file attached to.
     */
    fun jotId(): Long

    /**
     * Uri of this attachment.
     */
    fun uri(): String

    /**
     * Version of this attachment item.
     */
    fun version(): Int

    /**
     * Indicates if this attachment is deleted or not.
     */
    fun deleted(): Boolean
}